### ISSUE TITLE
log_ocs_version fails on cluster instaled with skip_ocs_deployment

### ISSUE THREAD

**Issue by dahorak** [red-hat-storage/ocs-ci#2592]
When only OCP cluster is installed (using `conf/ocsci/skip_ocs_deploy.yaml` config or `config.ENV_DATA['skip_ocs_deployment']` set to `True`), `log_ocs_version()` fixture tries to spin up ceph tool pod to get the version which fails, because OCS is not installed:

```
17:04:07 - MainThread - ocs_ci.deployment.deployment - WARNING - OCS deployment will be skipped
17:04:07 - MainThread - ocs_ci.ocs.version - INFO - collecting ocs version
17:04:07 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig ../clusters/2020-07-29T15-56-51_dahorak-flexy/auth/kubeconfig get clusterversion version -o yaml
17:04:07 - MainThread - ocs_ci.utility.utils - INFO - Executing command: git config --get user.name
17:04:07 - MainThread - ocs_ci.utility.utils - INFO - Executing command: git config --get user.email
17:04:07 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-console --kubeconfig ../clusters/2020-07-29T15-56-51_dahorak-flexy/auth/kubeconfig get  route -n openshift-console -o yaml
17:04:08 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig ../clusters/2020-07-29T15-56-51_dahorak-flexy/auth/kubeconfig get  nodes -o yaml
17:04:09 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig ../clusters/2020-07-29T15-56-51_dahorak-flexy/auth/kubeconfig get  clusterversion -n openshift-storage -o yaml
17:04:09 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig ../clusters/2020-07-29T15-56-51_dahorak-flexy/auth/kubeconfig get  clusterversion -n openshift-storage -o yaml
17:04:10 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig ../clusters/2020-07-29T15-56-51_dahorak-flexy/auth/kubeconfig get Pod  -n openshift-storage --selector=app=rook-ceph-tools -o yaml
17:04:10 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig ../clusters/2020-07-29T15-56-51_dahorak-flexy/auth/kubeconfig get pods -o name
17:04:11 - MainThread - ocs_ci.ocs.utils - INFO - starting ceph toolbox pod
17:04:11 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc patch ocsinitialization ocsinit -n openshift-storage --type json --patch  '[{ "op": "replace", "path": "/spec/enableCephTools", "value": true }]'
17:04:14 - MainThread - ocs_ci.utility.utils - WARNING - Command stderr: error: the server doesn't have a resource type "ocsinitialization"

ERROR                                                                                                                                                                                                                                  [100%]

=================================================================================================================== ERRORS ===================================================================================================================
_____________________________________________________________________________________________________ ERROR at setup of test_deployment ______________________________________________________________________________________________________

cluster = None

    @pytest.fixture(scope="session", autouse=True)
    def log_ocs_version(cluster):
        """
        Fixture handling version reporting for OCS.
    
        This fixture handles alignment of the version reporting, so that we:
    
         * report version for each test run (no matter if just deployment, just
           test or both deployment and tests are executed)
         * prevent conflict of version reporting with deployment/teardown (eg. we
           should not run the version logging before actual deployment, or after
           a teardown)
    
        Version is reported in:
    
         * log entries of INFO log level during test setup phase
         * ocs_version file in cluster path directory (for copy pasting into bug
           reports)
        """
        teardown = config.RUN['cli_params'].get('teardown')
        deploy = config.RUN['cli_params'].get('deploy')
        if teardown and not deploy:
            log.info("Skipping version reporting for teardown.")
            return
>       cluster_version, image_dict = get_ocs_version()

tests/conftest.py:217: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
ocs_ci/ocs/version.py:114: in get_ocs_version
    cluster_version['ENV'] = get_environment_info()
ocs_ci/ocs/version.py:76: in get_environment_info
    results['ceph_version'] = utils.get_ceph_version()
ocs_ci/utility/utils.py:1026: in get_ceph_version
    ct_pod = pod.get_ceph_tools_pod()
ocs_ci/ocs/resources/pod.py:457: in get_ceph_tools_pod
    setup_ceph_toolbox()
ocs_ci/ocs/utils.py:686: in setup_ceph_toolbox
    run_cmd(
ocs_ci/utility/utils.py:407: in run_cmd
    completed_process = exec_cmd(cmd, secrets, timeout, ignore_error, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cmd = ['oc', 'patch', 'ocsinitialization', 'ocsinit', '-n', 'openshift-storage', ...], secrets = None, timeout = 600, ignore_error = False, kwargs = {}
masked_cmd = 'oc patch ocsinitialization ocsinit -n openshift-storage --type json --patch  \'[{ "op": "replace", "path": "/spec/enableCephTools", "value": true }]\''
completed_process = CompletedProcess(args=['oc', 'patch', 'ocsinitialization', 'ocsinit', '-n', 'openshift-storage', '--type', 'json', '--...: true }]'], returncode=1, stdout=b'', stderr=b'error: the server doesn\'t have a resource type "ocsinitialization"\n')
masked_stdout = '', masked_stderr = 'error: the server doesn\'t have a resource type "ocsinitialization"\n'

    def exec_cmd(cmd, secrets=None, timeout=600, ignore_error=False, **kwargs):
        """
        Run an arbitrary command locally
    
        Args:
            cmd (str): command to run
            secrets (list): A list of secrets to be masked with asterisks
                This kwarg is popped in order to not interfere with
                subprocess.run(``**kwargs``)
            timeout (int): Timeout for the command, defaults to 600 seconds.
            ignore_error (bool): True if ignore non zero return code and do not
                raise the exception.
    
        Raises:
            CommandFailed: In case the command execution fails
    
        Returns:
            (CompletedProcess) A CompletedProcess object of the command that was executed
            CompletedProcess attributes:
            args: The list or str args passed to run().
            returncode (str): The exit code of the process, negative for signals.
            stdout     (str): The standard output (None if not captured).
            stderr     (str): The standard error (None if not captured).
    
        """
        masked_cmd = mask_secrets(cmd, secrets)
        log.info(f"Executing command: {masked_cmd}")
        if isinstance(cmd, str):
            cmd = shlex.split(cmd)
        completed_process = subprocess.run(
            cmd,
            stdout=subprocess.PIPE,
            stderr=subprocess.PIPE,
            stdin=subprocess.PIPE,
            timeout=timeout,
            **kwargs
        )
        masked_stdout = mask_secrets(completed_process.stdout.decode(), secrets)
        if len(completed_process.stdout) > 0:
            log.debug(f"Command stdout: {masked_stdout}")
        else:
            log.debug("Command stdout is empty")
    
        masked_stderr = mask_secrets(completed_process.stderr.decode(), secrets)
        if len(completed_process.stderr) > 0:
            log.warning(f"Command stderr: {masked_stderr}")
        else:
            log.debug("Command stderr is empty")
        log.debug(f"Command return code: {completed_process.returncode}")
        if completed_process.returncode and not ignore_error:
>           raise CommandFailed(
                f"Error during execution of command: {masked_cmd}."
                f"\nError is {masked_stderr}"
            )
E           ocs_ci.ocs.exceptions.CommandFailed: Error during execution of command: oc patch ocsinitialization ocsinit -n openshift-storage --type json --patch  '[{ "op": "replace", "path": "/spec/enableCephTools", "value": true }]'.
E           Error is error: the server doesn't have a resource type "ocsinitialization"

ocs_ci/utility/utils.py:461: CommandFailed
```

**Comment by mbukatov** [comment]
This is caused by https://github.com/red-hat-storage/ocs-ci/commit/487bdadfc9228294535beb115555611e7a93473f

**Comment by mbukatov** [comment]
I suggest to remove `cluster_version['ENV'] = get_environment_info()` line from `get_ocs_version()` function.

**Comment by mbukatov** [comment]
This issue is a deployment regression.

**Comment by Avilir** [comment]
PR was merge

**Comment by Avilir** [comment]
fixed with #2597
